### PR TITLE
CodeWhisperer: Add telemetry optout status to SendTelemetryEventRequest

### PR DESCRIPTION
**Pull request by andrewyuq**
Updated the API and put the field in SendTelemetryEvent API requests.

Corresponding VSC PR: https://github.com/aws/aws-toolkit-vscode/pull/3856

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by Will-ShaoHua — APPROVED** [review]

**Comment by andrewyuq** [comment]
Will fix the tests at a later time

**Comment by andrewyuq** [comment]
./retryBuilds

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/services/codewhisperer/CodeWhispererClientAdaptorTest.kt [hunk 1]
```diff
@@ -110,6 +118,13 @@ class CodeWhispererClientAdaptorTest {
             } doReturn authManagerRule.createConnection(ManagedSsoProfile("us-east-1", aString(), emptyList())) as AwsBearerTokenConnection
         }
         projectRule.project.replaceService(ToolkitConnectionManager::class.java, connectionManager, disposableRule.disposable)
+
+        isTelemetryEnabledDefault = AwsSettings.getInstance().isTelemetryEnabled
+    }
+
+    @After
+    fun tearDown() {
+        AwsSettings.getInstance().isTelemetryEnabled = isTelemetryEnabledDefault
```
Comment (manodnyab) [inline jetbrains-core/tst/software/aws/toolkits/jetbrains/services/codewhisperer/CodeWhispererClientAdaptorTest.kt:127]: Is this required? Are we changing the original value anywhere in the tests?

Comment (andrewyuq) [reply jetbrains-core/tst/software/aws/toolkits/jetbrains/services/codewhisperer/CodeWhispererClientAdaptorTest.kt:127]: yes here: 
```
AwsSettings.getInstance().isTelemetryEnabled = true
        mockApiCall()
        verify(bearerClient).sendTelemetryEvent(
            argThat<SendTelemetryEventRequest> { optOutPreference() == OptOutPreference.OPTIN }
        )

        AwsSettings.getInstance().isTelemetryEnabled = false
```

**Review by manodnyab — APPROVED** [review]

**Pull request merged**